### PR TITLE
Implement ADD and REMOVE changers to ExprMetadata

### PR DESCRIPTION
**Pull request by UnderscoreTud**
### Description
This PR adds `add` and `remove` changers to the metadata value expression, similar to the behaviour of adding and removing values from variables.

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/expressions/ExprMetadata.java [hunk 1]
```diff
@@ -38,101 +28,139 @@
...
+			this.keys = source.keys;
 			this.holders = source.holders;
 		}
 		this.types = types;
+		//noinspection unchecked
 		this.superType = (Class<T>) Utils.getSuperType(types);
 	}
 
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
```
Comment (Efnilite) [inline src/main/java/ch/njol/skript/expressions/ExprMetadata.java:93]:
```suggestion
	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
```

File: src/main/java/ch/njol/skript/expressions/ExprMetadata.java [hunk 2]
```diff
@@ -18,18 +18,8 @@
  */
```
Comment (Efnilite) [inline src/main/java/ch/njol/skript/expressions/ExprMetadata.java:18]: can be removed

**Review by Efnilite — APPROVED** [review]
nice

**Review by Efnilite — APPROVED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprMetadata.java [hunk 3]
```diff
@@ -151,8 +162,8 @@ public Expression<?> getSource() {
 	}
 
 	@Override
-	public String toString(@Nullable Event e, boolean debug) {
-		return "metadata values " + values.toString(e, debug) + " of " + holders.toString(e, debug);
+	public String toString(@Nullable Event event, boolean debug) {
+		return "metadata values " + keys.toString(event, debug) + " of " + holders.toString(event, debug);
 	}
 
 }
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/expressions/ExprMetadata.java:169]:
```suggestion
}

```

**Review by sovdeeth — APPROVED** [review]

**Review by cheeezburga — APPROVED** [review]

File: src/main/java/ch/njol/skript/expressions/ExprMetadata.java [hunk 4]
```diff
@@ -1,138 +1,148 @@
...
+			for (String key : keys) {
+                switch (mode) {
+                    case SET -> holder.setMetadata(key, new FixedMetadataValue(Skript.getInstance(), delta[0]));
+					case ADD, REMOVE -> {
+						assert delta != null;
+						Operator operator = mode == ChangeMode.ADD ? Operator.ADDITION : Operator.SUBTRACTION;
+						List<MetadataValue> metadata = holder.getMetadata(key);
+						Object value;
+						OperationInfo<?, ?, ?> info;
+						if (metadata.isEmpty() || (value = metadata.get(metadata.size() - 1).value()) == null) {
```
Comment (Pikachu920) [inline src/main/java/ch/njol/skript/expressions/ExprMetadata.java:121]: can we please not do the assignment here

File: src/test/skript/tests/syntaxes/expressions/ExprMetadata.sk [hunk 5]
```diff
@@ -0,0 +1,21 @@
+test "metadata value":
```
Comment (Pikachu920) [inline src/test/skript/tests/syntaxes/expressions/ExprMetadata.sk:1]: could we please get some tests for adding/removing/setting to incorrect typed and null values

**Review by Pikachu920 — CHANGES_REQUESTED** [review]

**Review by Moderocky — APPROVED** [review]

**Review by Pikachu920 — APPROVED** [review]

**Pull request merged**